ghcide/src/Development/IDE/Core/FileExists.hs

```diff
@@ -106,11 +107,11 @@ getFileExistsMapUntracked = do
   return v
 
 -- | Modify the global store of file exists.
-modifyFileExists :: IdeState -> [(NormalizedFilePath, FileChangeType)] -> IO ()
+modifyFileExists :: IdeState -> [(NormalizedFilePath, FileChangeType)] -> IO [Key]
 modifyFileExists state changes = do
   FileExistsMapVar var <- getIdeGlobalState state
   -- Masked to ensure that the previous values are flushed together with the map update
-  join $ mask_ $ atomicallyNamed "modifyFileExists" $ do
+  keys <- join $ mask_ $ atomicallyNamed "modifyFileExists" $ do
```

Comment:
we don't need the `join` if we get rid of the double return.

Reply:
right done